samples/other/file-transfer-http-to-http/transfer-file/src/main/java/com/siemens/mindsphere/datalake/edc/http/TransferExtension.java

```diff
@@ -0,0 +1,116 @@
+/*
+ *  Copyright (c) 2021, 2022 Siemens AG
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package com.siemens.mindsphere.datalake.edc.http;
+
+import org.eclipse.dataspaceconnector.dataloading.AssetLoader;
+import org.eclipse.dataspaceconnector.dataloading.ContractDefinitionLoader;
+import org.eclipse.dataspaceconnector.policy.model.Action;
+import org.eclipse.dataspaceconnector.policy.model.Permission;
+import org.eclipse.dataspaceconnector.policy.model.Policy;
+import org.eclipse.dataspaceconnector.spi.EdcSetting;
+import org.eclipse.dataspaceconnector.spi.asset.AssetSelectorExpression;
+import org.eclipse.dataspaceconnector.spi.asset.DataAddressResolver;
+import org.eclipse.dataspaceconnector.spi.policy.store.PolicyStore;
+import org.eclipse.dataspaceconnector.spi.security.Vault;
+import org.eclipse.dataspaceconnector.spi.system.Inject;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtension;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtensionContext;
+import org.eclipse.dataspaceconnector.spi.transfer.flow.DataFlowManager;
+import org.eclipse.dataspaceconnector.spi.transfer.inline.DataOperatorRegistry;
+import org.eclipse.dataspaceconnector.spi.types.domain.DataAddress;
+import org.eclipse.dataspaceconnector.spi.types.domain.asset.Asset;
+import org.eclipse.dataspaceconnector.spi.types.domain.contract.offer.ContractDefinition;
+import org.eclipse.dataspaceconnector.transfer.core.inline.InlineDataFlowController;
+
+import java.util.UUID;
+
+public class TransferExtension implements ServiceExtension {
+    @EdcSetting
+    private static final String STUB_URL = "edc.demo.http.source.url";
+
+    @Inject
+    private DataFlowManager dataFlowMgr;
+    @Inject
+    private DataAddressResolver dataAddressResolver;
+    @Inject
+    private DataOperatorRegistry dataOperatorRegistry;
+    @Inject
+    private ContractDefinitionLoader contractDefinitionLoader;
+    @Inject
+    private PolicyStore policyStore;
+    @Inject
+    private AssetLoader assetLoader;
+
+    @Override
+    public void initialize(ServiceExtensionContext context) {
+        var vault = context.getService(Vault.class);
+        var monitor = context.getMonitor();
+
+        dataOperatorRegistry.registerWriter(new HttpWriter(monitor));
+        dataOperatorRegistry.registerReader(new HttpReader(monitor));
+
+        dataFlowMgr.register(new InlineDataFlowController(vault, context.getMonitor(), dataOperatorRegistry));
+
+        String assetId = registerDataEntries(context);
+        monitor.info("Register http sample Asset: " + assetId);
+
+        var policy = createPolicy(assetId);
```

Comment:
are the sample policy/demo asset and contract definition still needed?

Reply:
I guess not. It was nice to get some validation with a dummy offer after fresh deployment due to in memory persistence. I propose to remove it after switch to Database persistence.